lib/grape/middleware/base.rb

```diff
@@ -21,9 +21,17 @@ def call(env)
 
       def call!(env)
         @env = env
+        error_class = nil
         before
-        @app_response = @app.call(@env)
-        after || @app_response
+        begin
+          @app_response = @app.call(@env)
+        rescue => e
+          error_class = e
+        ensure
+          after_response = after
+        end
+
+        after_response || (error_class ? (fail error_class) : @app_response)
       end
```

Comment:
Either all the paths aren't covered by specs or this can be:

```ruby
begin
  @app_response = @app.call(@env)
ensure
  after || @app_response
end
```
Causes the after middleware to execute on error, and raises the exception as before. 

What am I missing?